### PR TITLE
WIP ADD support for StackViewController to scroll to a child view controller

### PR DESCRIPTION
**Pull request by AndreasThenn**

### CODE REVIEW

File: Source/Views/StackViewController.swift [hunk 1]
```diff
@@ -9,6 +9,10 @@
 import UIKit
 import SnapKit
 
+public enum RelativePosition {
+    case topOrLeft, center, bottomOrRight
```
Comment (MP0w) [inline Source/Views/StackViewController.swift]: split case in multiple lines?
BTW instead of topOrLeft / bottomOrRight maybe beginning end? not sure

File: Source/Views/StackViewController.swift [hunk 2]
```diff
@@ -69,4 +73,58 @@ final class StackViewController: UIViewController {
         stackView.addArrangedSubview(childViewController.view)
         childViewController.didMove(toParentViewController: self)
     }
+    
+    /// A method to access the scrollView from outside and scroll to a given target ViewController. 
+    /// If no relativePosition parameter is provided, it will try to center the target.
+    ///
+    /// - Parameters:
+    ///   - target: Each child should have a meta representing `TabConfig`
```
Comment (MP0w) [inline Source/Views/StackViewController.swift]: copy & paste left over 😬

File: Source/Views/StackViewController.swift [hunk 3]
```diff
@@ -69,4 +73,58 @@ final class StackViewController: UIViewController {
...
+    ///   - relativePosition: An optional parameter to specify the alignement of the target. Depending of 
+    ///     the stackViews' orientation the relativePosition values have different meaning, e.g. top or 
+    ///     left.
+    public func scrollToViewController(target targetViewController: UIViewController,
```
Comment (MP0w) [inline Source/Views/StackViewController.swift]: as you mentioned this won't work for nested stacks you need to use  `func convert(CGRect, to: UIView?)` to convert a view frame to the equivalent in this VC's view.
However this will be enough for example for vertical stacks with vertical stacks but might break for example with a vertical stack that contains an horizontal and needs to be scrolled to:
- the right point vertically in the main stack
- the right point horizontally in the inner stack

I have a solution in mind but I need to write down some pseudo code,  maybe you guys can go ahead and eventually ping me if need help (I'm a bit busy and I might forget about it, let's see)

**Review by MP0w — CHANGES_REQUESTED** [review]
To test it you can snapshot it and consider all the edge cases:
- simple case
- nested simple (vertical w/ vertical)
- nested simple (horizontal w/ horizontal)
- nested complex (multiple horizontal and vertical)

File: Source/Views/StackViewController.swift (hunk 1 above)
Comment (AndreasThenn) [reply Source/Views/StackViewController.swift]: yeah, thx - I like your naming better

File: Source/Views/StackViewController.swift (hunk 2 above)
Comment (AndreasThenn) [reply Source/Views/StackViewController.swift]: puh, shame on me

File: Source/Views/StackViewController.swift (hunk 3 above)
Comment (AndreasThenn) [reply Source/Views/StackViewController.swift]: Thanks for the input. I guess at the moment, the current solution is working for us. But we probably come back to this by a more generic way.

**Comment by AndreasThenn** [comment]
I will close this PR until its ready to merge with working tests
